### DISCUSSION TITLE
[IMP] Component pagination

### DISCUSSION THREAD

**Discussion by Shide** [WeblateOrg/weblate#6034] [Ideas] [answered]
Every time a project loads on the screen, all compoents and his metrics are loaded and fetched too.

Will be awesome having a pager on components (nº of components to show per page would be configurable on the project) and the ability to have a group inside the Project>Search menu that allows to search and display components.

I've facing a lot of issues because I have several projects with thousands of components (Projects and Components are created via API).
These suggestion comes from a solution that I've implemented to translate a DDBB. Project are a DDBB table, records are components, and context are column names. This is the approach I've used to DDBB translation and it's perfect until I reach 6k records or so. Navigation on Weblate turns impossible mission and the magic of Weblate dissapears.

Weblate is awesome and I think the component pagination will give this project a lot of flexibility and will be capable of everything. Not just web translation.

**Comment by nijel** [comment, marked as answer]
The pagination is already there for some releeases, maybe you are using older version?

**Reply by Shide** [reply]
Maybe, I'm using 4.5 on production and 4.6 for development purposes

**Reply by nijel** [reply]
Then it's a good reason to upgrade, it was introduced in b732a09207c001218a0db13063fd1e25acb467c8, see #5513.

**Reply by Shide** [reply]
Marvelous Nijel, you are the best!
